packages/core/src/2d/assembler/TiledSpriteAssembler.ts

```diff
@@ -0,0 +1,383 @@
+import { Sprite } from "../sprite";
+import { SpriteRenderer } from "../sprite/SpriteRenderer";
+import { IAssembler } from "./IAssembler";
+import { Logger } from "../../base";
+
+/**
+ * @internal
+ */
+@StaticInterfaceImplement<IAssembler>()
+export class TiledSpriteAssembler {
+  static _worldMatrix: Matrix = new Matrix();
+  static _posRow: DisorderedArray<number> = new DisorderedArray<number>();
+  static _posColumn: DisorderedArray<number> = new DisorderedArray<number>();
+  static _uvRow: DisorderedArray<number> = new DisorderedArray<number>();
+  static _uvColumn: DisorderedArray<number> = new DisorderedArray<number>();
+
+  static resetData(renderer: SpriteRenderer): void {
+    renderer._verticesData.triangles ||= [];
+  }
+
+  static updatePositions(renderer: SpriteRenderer): void {
+    const { width, height, sprite, tileMode, tiledAdaptiveThreshold: threshold } = renderer;
+    const { positions, uvs, triangles } = renderer._verticesData;
+    // Calculate row and column
+    const { _posRow: posRow, _posColumn: posColumn, _uvRow: uvRow, _uvColumn: uvColumn } = this;
+    posRow.length = posColumn.length = uvRow.length = uvColumn.length = 0;
+    tileMode === SpriteTileMode.Adaptive
+      ? this._calculateAdaptiveDividing(sprite, width, height, threshold, posRow, posColumn, uvRow, uvColumn)
+      : this._calculateContinuousDividing(sprite, width, height, posRow, posColumn, uvRow, uvColumn);
+    // Update renderer's worldMatrix
+    const { x: pivotX, y: pivotY } = renderer.sprite.pivot;
+    const localTransX = renderer.width * pivotX;
+    const localTransY = renderer.height * pivotY;
+    // Renderer's worldMatrix
+    const { _worldMatrix: worldMatrix } = TiledSpriteAssembler;
+    const { elements: wE } = worldMatrix;
+    // Parent's worldMatrix
+    const { elements: pWE } = renderer.entity.transform.worldMatrix;
+    const sx = renderer.flipX ? -1 : 1;
+    const sy = renderer.flipY ? -1 : 1;
+    let wE0: number, wE1: number, wE2: number;
+    let wE4: number, wE5: number, wE6: number;
+    (wE0 = wE[0] = pWE[0] * sx), (wE1 = wE[1] = pWE[1] * sx), (wE2 = wE[2] = pWE[2] * sx);
+    (wE4 = wE[4] = pWE[4] * sy), (wE5 = wE[5] = pWE[5] * sy), (wE6 = wE[6] = pWE[6] * sy);
+    (wE[8] = pWE[8]), (wE[9] = pWE[9]), (wE[10] = pWE[10]);
+    const wE12 = (wE[12] = pWE[12] - localTransX * wE[0] - localTransY * wE[4]);
+    const wE13 = (wE[13] = pWE[13] - localTransX * wE[1] - localTransY * wE[5]);
+    const wE14 = (wE[14] = pWE[14] - localTransX * wE[2] - localTransY * wE[6]);
+    // Assemble position and uv
+    const rowLength = posRow.length - 1;
+    const columnLength = posColumn.length - 1;
+    let count = 0;
+    let trianglesOffset = 0;
+    for (let j = 0; j < columnLength; j++) {
+      for (let i = 0; i < rowLength; i++) {
+        const uvL = uvRow.get(2 * i);
+        const uvB = uvColumn.get(2 * j);
+        const uvR = uvRow.get(2 * i + 1);
+        const uvT = uvColumn.get(2 * j + 1);
```

Comment:
2 * i and 2 * j repeat

Reply:
I saved more than two calculations as temporary variables.